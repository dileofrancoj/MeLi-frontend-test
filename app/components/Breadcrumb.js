import React from "react";
import { Link } from "react-router-dom";
import { Row, Col } from "react-bootstrap";
import PropTypes from "prop-types";

const iterateCategories = (categories) => {
  return categories.map((cat, i) => (
    <Link to="" key={i}>
      <React.Fragment>
        {cat} {i < categories.length - 1 ? " > " : ""}
      </React.Fragment>
    </Link>
  ));
};

const Breadcrumb = ({ categories }) => {
  return (
    <Row className="mt-16 mb-16">
      <Col md={12}>
        {/* Se deja el Link por si se amplia a busqueda de categorias */}
        <span>{categories ? iterateCategories(categories) : null}</span>
      </Col>
    </Row>
  );
};

Breadcrumb.propTypes = {
  categories: PropTypes.array,
};

export default Breadcrumb;
