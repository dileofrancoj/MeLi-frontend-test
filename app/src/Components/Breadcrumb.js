import React from "react";
import { Row, Col } from "react-bootstrap";
import PropTypes from "prop-types";

const iterateCategories = (categories) => {
  return categories.map((cat, i) => (
    <span key={i}>
      <React.Fragment>
        {cat} {i < categories.length - 1 ? " > " : ""}
      </React.Fragment>
    </span>
  ));
};

const Breadcrumb = ({ categories }) => {
  return (
    <Row className="mt-16 mb-16">
      <Col md={12}>
        {/* Se deja el Link por si se amplia a busqueda de categorias */}
        <span>{categories ? iterateCategories(categories) : null}</span>
      </Col>
    </Row>
  );
};

Breadcrumb.propTypes = {
  categories: PropTypes.array,
};

export default Breadcrumb;
