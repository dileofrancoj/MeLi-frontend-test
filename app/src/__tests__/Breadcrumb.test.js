import React from "react";
import { shallow } from "enzyme";
import Breadcrumb from "./../Breadcrumb";

describe("Breadcrumb", () => {
  test("should render Breadcrumb component ", () => {
    const component = shallow(<Breadcrumb />);
    expect(component).toMatchSnapshot();
  });
  test("should show span element ", () => {
    const component = shallow(<Breadcrumb categories={["data"]} />);
    const element = component.find("span");
    expect(!!element).toBe(true);
    //expect(component).toMatcSnapshot();
  });
});
