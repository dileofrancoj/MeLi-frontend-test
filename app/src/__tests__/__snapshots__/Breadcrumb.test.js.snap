// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should render Breadcrumb component  1`] = `
<[object Object]
  className="mt-16 mb-16"
  noGutters={false}
>
  <[object Object]
    md={12}
  >
    <span
      className="f-14"
    />
  </[object Object]>
</[object Object]>
`;
